src/OrchardCore.Modules/OrchardCore.AuditTrail/Drivers/AuditTrailSettingsDisplayDriver.cs

```diff
@@ -0,0 +1,158 @@
+            !await IsAuthorizedToManageAuditTrailSettingsAsync()
+                ? null
+                : Initialize<AuditTrailSettingsViewModel>("AuditTrailSettings_Edit", model =>
+                {
+                    var descriptors = _auditTrailManager.DescribeCategories();
+                    var eventSettings = settings.EventSettings;
+
+                    var categories = descriptors.GroupBy(categoryDescriptor => categoryDescriptor.Category)
+                        .Select(categoryDescriptor =>
+                        new
+                        {
+                            Category = categoryDescriptor.Key,
+                            Name = categoryDescriptor.Select(categoryDescriptor => categoryDescriptor.LocalizedName),
+                            Events = categoryDescriptor.SelectMany(categoryDescriptor => categoryDescriptor.Events)
+                        })
+                        .Select(categoryDescriptor =>
+                            new AuditTrailCategorySettingsViewModel
+                            {
+                                Category = categoryDescriptor.Category,
+                                Name = categoryDescriptor.Name.First(),
+                                Events = categoryDescriptor.Events.Select(eventDescriptor =>
+                                {
+                                    var eventSetting = GetOrCreate(eventSettings.ToList(), eventDescriptor);
+
+                                    return new AuditTrailEventSettingsViewModel
+                                    {
+                                        Event = eventDescriptor.FullEventName,
+                                        Name = eventDescriptor.LocalizedName,
+                                        Description = eventDescriptor.Description,
+                                        IsEnabled = eventDescriptor.IsMandatory || eventSetting.IsEnabled,
+                                        IsMandatory = eventDescriptor.IsMandatory
+                                    };
+                                }).ToList()
+                            }).ToList();
+
+                    model.Categories = categories;
+                    model.EnableClientIpAddressLogging = settings.EnableClientIpAddressLogging;
+                    model.IgnoredContentTypes = _contentDefitinionManager.ListTypeDefinitions()
+                        .Select(contentTypeDefiniton =>
+                            new IgnoredContentTypesViewModel
+                            {
+                                Name = contentTypeDefiniton.Name,
+                                DisplayName = contentTypeDefiniton.DisplayName,
+                                Selected = settings.IgnoredContentTypeNames.Contains(contentTypeDefiniton.Name)
+                            }).OrderBy(contentTypeDefinition => contentTypeDefinition.DisplayName).ToList();
```

Comment:
`OrderBy()` and `ToList()` could be on new lines each for clarity. Other similar occasions in this file too.